lib/Runtime/Library/TypedArray.cpp

```diff
@@ -144,10 +144,13 @@ namespace Js
                 if (JavascriptOperators::IsObject(firstArgument))
                 {
                     Var iterator = JavascriptOperators::GetProperty(RecyclableObject::FromVar(firstArgument), PropertyIds::_symbolIterator, scriptContext);
+                    Var arrayIteratorPrototypeNext = nullptr;
 
                     if (!JavascriptOperators::IsUndefinedObject(iterator) &&
                         (iterator != scriptContext->GetLibrary()->GetArrayPrototypeValuesFunction() ||
-                        !JavascriptArray::Is(firstArgument)))
+                            !JavascriptArray::Is(firstArgument) ||
+                            !RuntimeFunction::Is(arrayIteratorPrototypeNext = JavascriptOperators::GetProperty(scriptContext->GetLibrary()->GetArrayIteratorPrototype(), PropertyIds::next, scriptContext)) ||
```

Comment:
`JavascriptOperators::GetProperty` can execute a getter or proxy trap.  It can't be used here because of that unfortunately.

I don't know off the top of my head if we have an API to do a get that will get a property value without executing a trap nor getter.  That is what we need here though.

E.g. this code will behave badly:

```javascript
var iter = [][Symbol.iterator]();
var count = 0;
Object.defineProperty(iter.__proto__, 'next', {
    get: function () {
        count++;
        return function () {
            return { done: true };
        };
    }
});

var i32a = new Int32Array([1, 2, 3]);

for (let i = 0; i < i32a.length; i += 1) {
    print(i32a[i]);
}
print(count);
```

`count` will be one too large.